### PR TITLE
Block level-month report

### PR DESCRIPTION
**Pull request by agrzywinski**
@snopoke I've added a sample format 3 and mechanism to merge several reports into one.
About your comments to previous PR:
- I have to keep grouping in the ASHAFunctionalityChecklistData, without it query return only one row, but I changed case_id to doc_id and filter in the ASHAAFChecklistData
- Property "table_name" is required but I've removed unnecessary _qualify_column functions

### CODE REVIEW

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: custom/up_nrhm/reports/asha_reports.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import datetime
```
Comment (dimagimon) [inline custom/up_nrhm/reports/asha_reports.py]: F401 'datetime' imported but unused

File: custom/up_nrhm/reports/asha_reports.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+from dateutil.relativedelta import relativedelta
```
Comment (dimagimon) [inline custom/up_nrhm/reports/asha_reports.py]: F401 'relativedelta' imported but unused

File: custom/up_nrhm/reports/asha_reports.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+from corehq.apps.reports.filters.select import MonthFilter, YearFilter
+from corehq.apps.reports.generic import GenericTabularReport
+from corehq.apps.reports.standard import CustomProjectReport, DatespanMixin
+from corehq.apps.reports.filters.dates import DatespanFilter
+from custom.up_nrhm.filters import DrillDownOptionFilter, SampleFormatFilter
+from custom.up_nrhm.reports.asha_facilitators_report import ASHAFacilitatorsReport
+from custom.up_nrhm.reports.block_level_af import BlockLevelAFReport
```
Comment (dimagimon) [inline custom/up_nrhm/reports/asha_reports.py]: F401 'BlockLevelAFReport' imported but unused

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: custom/up_nrhm/reports/asha_reports.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+    @property
+    def headers(self):
+        return self.model.headers
+
+    @property
+    def rows(self):
+        config = self.report_config
+        if not config.get('sf'):
+            rows, self.total_under_facilitator, total_with_checklist = self.model.rows
```
Comment (snopoke) [inline custom/up_nrhm/reports/asha_reports.py]: I think this should be `self.total_with_checklist`

**Comment by snopoke** [comment]
Looks good apart from one comment

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by agrzywinski** [comment]
@snopoke

**Pull request merged**